pandas/_libs/tslibs/conversion.pyx

```diff
@@ -347,10 +347,9 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
     if tz is not None:
         tz = maybe_get_tz(tz)
 
-        # sort of a temporary hack
         if ts.tzinfo is not None:
             if hasattr(tz, 'normalize') and hasattr(ts.tzinfo, '_utcoffset'):
```

Comment:
This check might be superfluous since `datetimes` natively have the `astimezone` method. Then this whole block might be able to be simplified.

Reply:
hmm, you can try that and see if it works. maybe add a comment anyhow